File: app/src/main/java/com/victoria/bleled/common/receiver/InstallReceiver.java
package com.victoria.bleled.common.receiver;

import android.content.BroadcastReceiver;
import android.content.Context;
import android.content.Intent;
import android.util.Log;

import com.victoria.bleled.data.repository.DataStoreRepository;

import javax.inject.Inject;

public class InstallReceiver extends BroadcastReceiver {
    @Inject
    DataStoreRepository dataStoreRepository;

    public void onReceive(Context context, Intent intent) {
        String rawReferrerString = intent.getStringExtra("referrer");
        if (rawReferrerString != null) {
            Log.i("MyApp", "Received the following intent " + rawReferrerString);
            //dataStoreRepository.putString(DataStoreKey.PREFS_REFERER, rawReferrerString, GlobalScope.INSTANCE.);
        }
    }
}

File: app/src/main/java/com/victoria/bleled/common/InstallReceiver.kt
package com.victoria.bleled.common

import android.content.BroadcastReceiver
import android.content.Context
import android.content.Intent
import com.victoria.bleled.app.MyApplication
import com.victoria.bleled.data.local.pref.PrefDataSourceImpl

class InstallReceiver : BroadcastReceiver() {
    override fun onReceive(context: Context, intent: Intent) {
        val rawReferrerString = intent.getStringExtra("referrer")

        if (rawReferrerString != null) {
            val dataManager =
                PrefDataSourceImpl.getInstance(MyApplication.globalApplicationContext!!)
            dataManager.installReferer = rawReferrerString
        }
    }
}